### ISSUE TITLE
source.unsplash.com no longer working

### ISSUE THREAD

**Issue by lkthomas** [BrianHepler/MMM-WeatherBackground#24]
https://source.unsplash.com/collection/4733334/?winter,day,cloudy&s=1631542013371

![image](https://github.com/BrianHepler/MMM-WeatherBackground/assets/581053/92af3972-5ba2-4c40-9587-02e83697590a)

I think source.unsplash.com no longer work

**Comment by BrianHepler** [comment]
Hmm. Looking into it now.

**Comment by D3dl3g** [comment]
if it helps, reddit suggests that `source.unsplash.com` appears to be depreciated, Unsplash moved to API. 

looking forward to this hopefully getting an update. i love this Module @BrianHepler thank you for its development and not abandoning it, yet.

edit:- 
Reddit (https://www.reddit.com/r/unsplash/comments/s13x4h/what_happened_to_sourceunsplashcom/) gives these 3 links for further info 

- https://unsplash.com/developers
- https://unsplash.com/documentation
- https://help.unsplash.com/en/collections/1451694-api-guidelines

**Comment by BrianHepler** [comment]
I'm in talks with Unsplash to get an application license. They've been slow to respond.

**Comment by lkthomas** [comment]
Are they allowed to publish API token?

**Comment by BrianHepler** [comment]
No, they have a two-key system. Developer key & secret key. You're not supposed to share either one. I'm trying to get a response on having the users register and get an API key without having to register as developers. They operate from the perspective of people building mobile apps, not open source.

**Comment by kevinsivic** [comment]
I've created a pull request that requires users to register and get an API key.

**Comment by kiluazen** [comment]
`source.unsplash.com` was officially deprecated and shut down by Unsplash in 2024. There's no direct replacement for the random image URL pattern it provided.

If you're looking for a no-key alternative, [tteg](https://github.com/kiluazen/tteg) provides Unsplash image search without requiring API registration:

```bash
# Search and get image URLs
curl "https://tteg-api-53227342417.asia-south1.run.app/search?q=weather+landscape&n=1&orientation=landscape"

# Or install the CLI and save directly
uv tool install tteg
tteg save "rainy day cityscape" ./background.jpg
```

No Unsplash developer account needed. Free tier is 50 queries/day.

**Comment by BrianHepler** [comment]
That just might do it. Thank you. I'll investigate.
